neonvm/controllers/metrics.go

```diff
@@ -62,6 +70,32 @@ func MakeReconcilerMetrics() ReconcilerMetrics {
 				Help: "Total number of VM restarts across the cluster captured by VirtualMachine reconciler",
 			},
 		)),
+		reconcileDurationSuccessful: util.RegisterMetric(metrics.Registry, prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name:    "reconcile_duration_successful_seconds",
+				Help:    "Time duration of successful reconciles",
+				Buckets: buckets,
+			},
+		)),
+		reconcileDurationFailure: util.RegisterMetric(metrics.Registry, prometheus.NewHistogram(
+			prometheus.HistogramOpts{
+				Name:    "reconcile_duration_failure_seconds",
+				Help:    "Time duration of failed reconciles",
+				Buckets: buckets,
+			},
+		)),
```

Comment:
Hm, thinking about what usage of this will look like - getting total average reconcile duration would be something like `(rate(reconcile_duration_successful_seconds_sum{}) + rate(reconcile_duration_failure_seconds_sum{})) / (rate(reconcile_duration_successful_seconds_count{}) + rate(reconcile_duration_failure_seconds_count{}))`

What do you think about making this a single metric and then differentiate via `outcome="..."` ? Could bundle in the status failures as well with `outcome="conflict"`...